src/metabase/channel/template/handlebars.clj

```diff
@@ -0,0 +1,64 @@
+(ns metabase.channel.template.handlebars
+  (:require
+   [clojure.walk :as walk]
+   [metabase.util :as u])
+  (:import
+   (java.io IOException)
+   (com.github.jknack.handlebars
+    Handlebars Helper Template)
+   (com.github.jknack.handlebars.io
+    TemplateLoader ClassPathTemplateLoader CompositeTemplateLoader TemplateSource)
+   (com.github.jknack.handlebars.cache
+    ConcurrentMapTemplateCache)))
+
+(deftype AppDBTemplateSource [^String name ^String the-content last-modified]
+  TemplateSource
+  (content [_ _] the-content)
+  (filename [_] name)
+  (lastModified [_] last-modified))
+
+(deftype AppDBLoader []
+  TemplateLoader
+  (sourceAt [_this _template-name]
+    (throw (IOException. "Not implemented"))
+    #_(AppDBTemplateSource. template-name "hello {{name}}" (System/currentTimeMillis))
+    nil))
+
+(defn- composite-loaders [& loaders]
+  (CompositeTemplateLoader. (into-array TemplateLoader loaders)))
+
+(def ^:private handlebars
+  (doto (Handlebars. (composite-loaders (AppDBLoader.) (ClassPathTemplateLoader. "/" ".mustache")))
+    (.with (ConcurrentMapTemplateCache.))))
+
+(defn- wrap-context
+  [context]
+  (walk/postwalk
+   #(cond
+     (map? %) (java.util.HashMap. ^java.util.Map %)
```

Comment:
It's a real pity we need to copy the data like this. What's the interface it expects? Maybe we can save some memory with wrapper instead.